db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java

```diff
@@ -0,0 +1,95 @@
+package com.eventstore.dbclient;
+
+public abstract class AbstractPersistentSubscriptionSettings<T> {
+    private int checkpointAfterMs;
+    private boolean extraStatistics;
+    private boolean resolveLinks;
+    private boolean fromStart;
+    private boolean fromEnd;
+    private int historyBufferSize;
+    private int liveBufferSize;
+    private int maxCheckpointCount;
+    private int maxRetryCount;
+    private int maxSubscriberCount;
+    private int messageTimeoutMs;
+    private int minCheckpointCount;
+    private int readBatchSize;
+    private ConsumerStrategy strategy;
+
+    public AbstractPersistentSubscriptionSettings(int checkpointAfterMs, boolean extraStatistics, boolean resolveLinks,
+                                                  int historyBufferSize, int liveBufferSize, int maxCheckpointCount,
+                                                  int maxRetryCount, int maxSubscriberCount, int messageTimeoutMs,
+                                                  int minCheckpointCount, int readBatchSize, ConsumerStrategy strategy,
+                                                  boolean fromStart, boolean fromEnd) {
+        this.checkpointAfterMs = checkpointAfterMs;
+        this.extraStatistics = extraStatistics;
+        this.resolveLinks = resolveLinks;
+        this.historyBufferSize = historyBufferSize;
+        this.liveBufferSize = liveBufferSize;
+        this.maxCheckpointCount = maxCheckpointCount;
+        this.maxRetryCount = maxRetryCount;
+        this.maxSubscriberCount = maxSubscriberCount;
+        this.messageTimeoutMs = messageTimeoutMs;
+        this.minCheckpointCount = minCheckpointCount;
+        this.readBatchSize = readBatchSize;
+        this.fromStart = fromStart;
+        this.fromEnd = fromEnd;
+        this.strategy = strategy;
+    }
+
+    public int getCheckpointAfterMs() {
+        return checkpointAfterMs;
+    }
+
+    public boolean isExtraStatistics() {
+        return extraStatistics;
+    }
+
+    public boolean isResolveLinks() {
+        return resolveLinks;
+    }
+
+    public int getHistoryBufferSize() {
+        return historyBufferSize;
+    }
+
+    public int getLiveBufferSize() {
+        return liveBufferSize;
+    }
+
+    public int getMaxCheckpointCount() {
```

Comment:
**FIX:** Setting does not match the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R43).
```suggestion
    public int getCheckPointUpperBound() {
```
The same comment applies e.g. https://github.com/EventStore/EventStoreDB-Client-Java/pull/84/files#diff-b38c322fb2ef3a75dc3289f2babf28c7359937f0530d73c8f3a3734ac18b4dceR28.